src/wp-includes/kses.php

```diff
@@ -144,6 +146,7 @@
 			'align' => true,
 			'open'  => true,
 		),
+		'dialog'     => array(),
```

Comment:
The [`closedby`](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog#closedby) attribute should be added here at least.